File: LOO 2°Parcial/Tarea 4/EmpleadoPiezas.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package empleadospolimorfismo;

/**
 * 
 * @author Dimitri Medina Novelo
 */
public class EmpleadoPiezas extends Empleado {
        protected int cantidadPiezas;
        protected double pagoXPieza;

        public EmpleadoPiezas()
        {
            super ();
            this.cantidadPiezas = 0;
            this.pagoXPieza = 0;
        }

        public EmpleadoPiezas(String nombre, int cantidad, double pago) 
        {
            super(nombre);
            this.cantidadPiezas = cantidad > 0 ? cantidad : 0;
            this.pagoXPieza = pago > 0 ? pago : 0;
        }
        
        public int getCantidadPiezas()
        {
            return this.cantidadPiezas;
        }
        
        public void setCantidadPiezas(int cantidad)
        {
            this.cantidadPiezas = cantidad > 0 ? cantidad : 0;
        }
        
        public double getPagoXPieza()
        {
            return this.pagoXPieza;
        }
        
        public void setPagoXPieza(double pago)
        {
            this.pagoXPieza = pago > 0 ? pago : 0;
        }
        
        @Override
        public double pagoDiario()
        {
            return this.cantidadPiezas * this.pagoXPieza;
        }
        
        @Override
        public void imprimir()
        {
            System.out.println("Empleado tipo TRABAJADOR POR PIEZAS");
            System.out.println("NOMBRE: " + this.nombre + "\nPAGO POR PIEZA: " + this.pagoXPieza + "\nCANTIDAD DE PIEZAS: " + this.cantidadPiezas + "\nPAGO DIARIO: " + this.pagoDiario());
        }
}

File: LOO 2°Parcial/Tarea 3/EmpleadoPiezas.cpp
#include "EmpleadoPiezas.h"

EmpleadoPiezas::EmpleadoPiezas() : Empleado(){
	this->cantidadPiezas = 0;
	this->pagoXPieza = 0;
}
EmpleadoPiezas::EmpleadoPiezas(string nombre, int cantidad, double pago) : Empleado(nombre){
	this->cantidadPiezas = cantidad > 0 ? cantidad : 0;
	this->pagoXPieza = pago > 0 ? pago : 0;
}
int EmpleadoPiezas::getCantidadPiezas(void){
	return this->cantidadPiezas;
}
void EmpleadoPiezas::setCantidadPiezas(int cantidad){
	this->cantidadPiezas = cantidad > 0 ? cantidad : 0;
}
double EmpleadoPiezas::getPagoxPieza(void){ return this->pagoXPieza; }
void EmpleadoPiezas::setPagoxPieza(double pago){ this->pagoXPieza = pago > 0 ? pago : 0; }
double EmpleadoPiezas::pagoDiario(void){ return this->cantidadPiezas * this->pagoXPieza; }
void EmpleadoPiezas::imprimir(void){
	cout << "Empleado tipo TRABAJADOR POR PIEZAS" << endl;
	cout << "NOMBRE: " << this->nombre << "\nPAGO POR PIEZA: " << this->pagoXPieza << "\nCANTIDAD PIEZAS HECHAS: " << this->cantidadPiezas << "\nPAGO DIARIO: " << this->pagoDiario() << endl;
}
